frontend/context/StakingProgramsProvider.tsx

```diff
@@ -0,0 +1,87 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { createContext, PropsWithChildren, useCallback } from 'react';
+
+import { FIVE_SECONDS_INTERVAL } from '@/constants/intervals';
+import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
+import { StakingProgramId } from '@/enums/StakingProgram';
+import { useServiceId } from '@/hooks/useService';
+import { useServices } from '@/hooks/useServices';
+import { Nullable } from '@/types/Util';
+
+const INITIAL_DEFAULT_STAKING_PROGRAM_ID = StakingProgramId.PearlBeta;
+
+export const StakingProgramsContext = createContext<{
+  isActiveStakingProgramsLoaded: boolean;
+  activeStakingProgramsId: StakingProgramId[];
+}>({
+  isActiveStakingProgramsLoaded: false,
+  activeStakingProgramsId: [],
+});
+
+/**
+ * hook to get the active staking program id
+ */
+const useGetActiveStakingProgramIds = () => {
+  const queryClient = useQueryClient();
+  const { selectedAgentConfig } = useServices();
+  const serviceId = useServiceId();
+
+  const { serviceApi, homeChainId } = selectedAgentConfig;
+
+  const response = useQuery({
+    queryKey: REACT_QUERY_KEYS.STAKING_PROGRAM_KEY(homeChainId, serviceId!),
+    queryFn: async () => {
+      const response = await serviceApi.getCurrentStakingProgramsByServiceId(
+        serviceId!,
+        homeChainId,
+      );
+      return response?.length === 0
+        ? [INITIAL_DEFAULT_STAKING_PROGRAM_ID]
+        : response;
+    },
+    enabled: !!homeChainId && !!serviceId,
+    refetchInterval: serviceId ? FIVE_SECONDS_INTERVAL : false,
+  });
+
+  const setActiveStakingProgramId = useCallback(
+    (stakingProgramId: Nullable<StakingProgramId>) => {
+      if (!serviceId) return;
+      if (!stakingProgramId) return;
+
+      // update the active staking program id in the cache
+      queryClient.setQueryData(
```

Comment:
this won't update the response above automatically, it'll just update the cache which is just an object and doesn't work like a hook. if you want the response to update too, suggest using [mutation](https://tanstack.com/query/v4/docs/framework/react/guides/optimistic-updates#updating-a-single-todo) instead or other approaches, e.g. like we did in launch app - clearing the cache part by a key so it's re-fetched automatically